temporian/core/typing.py

```diff
@@ -64,5 +64,18 @@
 is equivalent to `{event_set.node() : event_set}`.
 """
 
-
 TypeOrDType = Union[DType, Type[float], Type[int], Type[str], Type[bool]]
+
+
+IndexKeyItem = Union[int, str, bytes]
```

Comment:
> str
I think this is the difference between user inputs  and internal representation.
We don't allow index on strings. If a user feeds a string (which is allowed), it is converted into a bytes.